### PR TITLE
added filling types

### PR DESCRIPTION
**Pull request by hircumg**

### CODE REVIEW

File: slicers/fill_layer.go [hunk 1]
```diff
@@ -7,16 +7,71 @@ import (
...
+
+	return planes
+}
+
+func CalcFillPlanes(mesh *Mesh, settings Settings) []Plane {
 	planes := []Plane{}
-	for i := 0; i < n; i++ {
-		curP = curP.Shift(AxisX.MulScalar(step))
-		planes = append(planes, Plane{curP, AxisX})
+	if settings.FillingType == "Squares" {
```
Comment (l1va) [inline slicers/fill_layer.go]: https://tour.golang.org/flowcontrol/9

File: slicers/fill_layer.go [hunk 2]
```diff
@@ -7,16 +7,71 @@ import (
...
+	planes := []Plane{}
+	planes = append(planes, Plane{curP, MyAxis})
+
+	for i := 0; i < nMax; i++ {
+		curP = curP.Shift(MyAxis.MulScalar(step))
+		planes = append(planes, Plane{curP, MyAxis})
+	}
+
+	curP = MyAxis.MulScalar(0).ToPoint()
+	for i := 0; i < nMin; i++ {
```
Comment (l1va) [inline slicers/fill_layer.go]: What if min > 0? nMin will be <0

File: slicers/fill_layer.go [hunk 3]
```diff
@@ -7,16 +7,71 @@ import (
...
+
+	planes := []Plane{}
+	planes = append(planes, Plane{curP, MyAxis})
+
+	for i := 0; i < nMax; i++ {
+		curP = curP.Shift(MyAxis.MulScalar(step))
+		planes = append(planes, Plane{curP, MyAxis})
+	}
+
+	curP = MyAxis.MulScalar(0).ToPoint()
```
Comment (l1va) [inline slicers/fill_layer.go]: MulScalar(0) return 0, is it right behaviour?

File: slicers/fill_layer.go [hunk 4]
```diff
@@ -7,16 +7,71 @@ import (
 	"sort"
 )
 
-func CalcFillPlanes(mesh *Mesh, settings Settings) []Plane {
-	minx, maxx := mesh.MinMaxZ(AxisX)
+func CalcFillPlanes0(mesh *Mesh, settings Settings, MyAxis Vector, step float64) []Plane {
```
Comment (l1va) [inline slicers/fill_layer.go]: maybe name it with 'common' word and make it hidden ( started with a small letter)

File: slicers/fill_layer.go (hunk 3 above)
Comment (hircumg) [reply slicers/fill_layer.go]: yes

File: slicers/fill_layer.go (hunk 1 above)
Comment (hircumg) [reply slicers/fill_layer.go]: done

File: slicers/fill_layer.go (hunk 2 above)
Comment (hircumg) [reply slicers/fill_layer.go]: done

File: cmd/goosli/goosli.go [hunk 5]
```diff
@@ -35,6 +35,7 @@ var (
 	printSpeedWall      = kingpin.Flag("print_speed_wall", "Printing speed of walls.").Default("50").Int()
 	fanOffLayer1        = kingpin.Flag("fan_off_layer1", "Turn off the fan for Layer 1.").Bool()
 	nozzle              = kingpin.Flag("nozzle", "Nozzle diameter.").Default("0.4").Float64()
+	fillingType         = kingpin.Flag("filling_type", "Filling type(Lines,Squares,Triangles)").Default("Lines").Int()
```
Comment (l1va) [inline cmd/goosli/goosli.go]: does your default works with int?

**Pull request merged**